### PR TITLE
fix(file-uploader): match experimental spec

### LINKED ISSUE #1746: Experimental file uploader does not match spec
The experimental file uploader appears to be missing error states and some default styling

Related:

https://github.com/IBM/carbon-components/issues/1255

https://github.com/IBM/carbon-components/issues/1296

https://github.com/IBM/carbon-components/pull/1450

https://github.com/IBM/carbon-components/issues/1535

### PR DESCRIPTION
**Pull request by emyarod**
Closes #1746

This PR modifies the experimental file uploader to match the spec from https://github.com/IBM/carbon-components/issues/1255 and https://github.com/IBM/carbon-components/issues/1535

#### Changelog

**New**

- show file upload states on component docs
- show example error state

**Changed**

- experimental component spacing and colors

### CODE REVIEW

File: src/components/file-uploader/_file-uploader.scss [hunk 1]
```diff
@@ -152,13 +161,42 @@
     align-items: center;
     justify-content: space-between;
     min-height: rem(30px);
-    background-color: $inverse-01;
+    background-color: $ibm-colors__gray-10;
```
Comment (asudoh) [inline src/components/file-uploader/_file-uploader.scss]: @laurenmrice Is there a theme token that should be applied to selected file's background color? Thanks!

File: src/components/file-uploader/_file-uploader.scss [hunk 2]
```diff
@@ -152,13 +161,42 @@
...
+      right: -0.25rem; // offset due to larger SVG size
+    }
+  }
+
+  .#{$prefix}--file__selected-file--invalid {
+    @include focus-outline('invalid');
+    margin-bottom: $spacing-2xs;
+
+    .#{$prefix}--file-filename {
+      color: $ibm-colors__gray-30;
```
Comment (asudoh) [inline src/components/file-uploader/_file-uploader.scss]: @laurenmrice Is there a theme token that should be applied to invalid filename's color? Thanks!

**Review by asudoh — COMMENTED** [review]
Just a couple of comments - Thanks @emyarod!

**Review by tw15egan — APPROVED** [review]
Looks good once comments are addressed

File: src/components/file-uploader/_file-uploader.scss (hunk 2 above)
Comment (laurenmrice) [reply src/components/file-uploader/_file-uploader.scss]: Hey Akira! The type style token is caption-01 and the color token for the invalid text is support-01.

File: src/components/file-uploader/_file-uploader.scss (hunk 1 above)
Comment (laurenmrice) [reply src/components/file-uploader/_file-uploader.scss]: The background color of the upload should be color token field-01. There is not a selected state for the uploaded file.

Comment (emyarod) [reply src/components/file-uploader/_file-uploader.scss]: @laurenmrice just to clarify, I believe this is not for a selected state. It just refers to the file(s) currently stored in the uploader input

Comment (laurenmrice) [reply src/components/file-uploader/_file-uploader.scss]: Oh okay yeah, all of the files that have been uploaded should all have a background of field-01 (#F3F3F3). When you get a chance can you update the link, it is broken for me :)

Comment (emyarod) [reply src/components/file-uploader/_file-uploader.scss]: @laurenmrice just fixed it, the link should be working again now

**Review by aledavila — APPROVED** [review]
lgtm - just viz

**Comment by asudoh** [comment]
@aagonzales @jeanservaas Could either of you review the viz on behalf of @laurenmrice...? Thanks!

**Comment by jeanservaas** [comment]
Hey guys,

I noticed that file uploader was not pulling in the primary button (@aagonzales we may need to check the kit here too) 

- the button should have 64px fixed right padding after type

- There should be 24px of space between the button and the first field (under the button)

- the type in the fields should be 14px not 12px (they should use the token $body-short-01 

- the fields should be 32px high (same as button) -- they are inspecting at 30 now (maybe that's why the icons look small)

image

**Comment by emyarod** [comment]
@jeanservaas the latest fixes should now be on the netlify link

**Review by tw15egan — APPROVED** [review]
Looks great!

**Review by jeanservaas — APPROVED** [review]

**Pull request merged**